HelperFunctions.ps1

```diff
@@ -987,39 +987,68 @@ function DownloadFileLow {
         }
     }
     else {
-        Write-Host "Downloading using HttpClient"
-        if ($useDefaultCredentials) {
-            $handler = New-Object System.Net.Http.HttpClientHandler
-            $handler.UseDefaultCredentials = $true
-            $httpClient = New-Object System.Net.Http.HttpClient -ArgumentList $handler
-        }
-        else {
-            $httpClient = New-Object System.Net.Http.HttpClient
-        }
-        $httpClient.Timeout = [Timespan]::FromSeconds($timeout)
-        $headers.Keys | ForEach-Object {
-            $httpClient.DefaultRequestHeaders.Add($_, $headers."$_")
-        }
-        $stream = $null
-        $fileStream = $null
-        if ($dontOverwrite) {
-            $fileMode = [System.IO.FileMode]::CreateNew
+        if ($PSVersionTable.PSVersion -ge "7.0.0") {
+            Write-Host "Downloading using Invoke-WebRequest"
+            if ($useDefaultCredentials) {
+                $response = Invoke-WebRequest -UseDefaultCredentials -TimeoutSec $timeout -Headers $headers -Uri $sourceUrl -SkipCertificateCheck
+            }
+            else {
+                $response = Invoke-WebRequest -TimeoutSec $timeout -Headers $headers -Uri $sourceUrl -SkipCertificateCheck

```

Comment:
isn't -useBasicParsing needed as well?

Reply:
Non longer with PowerShell 7, according to the documentation - it is native behavior since PowerShell 6

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.3#-usebasicparsing